metricflow/sql/sql_exprs.py

```diff
@@ -786,6 +791,98 @@ def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D
         return self.sql_function == other.sql_function and self._parents_match(other)
 
 
+class SqlPercentileFunctionType(Enum):
+    """Type of percentile function used."""
+
+    DISCRETE = "discrete"
+    CONTINUOUS = "continuous"
+
+
+@dataclass(frozen=True)
+class SqlPercentileExpressionArgument:
+    """Dataclass for holding percentile arguments"""
+
+    percentile: float
+    function_type: SqlPercentileFunctionType
+
+    @property
+    def function_name(self) -> str:
+        """Helper to build function name to use in rendering"""
+        return "PERCENTILE_CONT" if self.function_type == SqlPercentileFunctionType.CONTINUOUS else "PERCENTILE_DISC"
+
+
+class SqlPercentileExpression(SqlFunctionExpression):
+    """An percentile aggregation expression."""
```

Comment:
micro-nit: 
```suggestion
    """A percentile aggregation expression."""
```

Reply:
For sure! Will merge after the query_parser pr goes through. Think I neglected to rebase before merging today, my bad!